### PR TITLE
Profuse tea: handle join team tokens

### PR DESCRIPTION
**Pull request by GregWeil**
I think this could be done better, but I'm not sure how. It turns out we can't just add you to the team single-page-app-ly because then you miss out on admin info and anything else that doesn't load if you weren't already a team member.

I also removed querystringify in favor of the standard `new URLSearchParams()`

### CODE REVIEW

File: src/presenters/pages/team.jsx [hunk 1]
```diff
@@ -266,6 +269,47 @@ TeamPage.propTypes = {
...
+  async componentDidMount() {
+    const params = new URLSearchParams(this.props.location.search);
+    if (params.has('joinToken')) {
+      try {
+        await this.props.api.post(`/teams/${this.props.team.id}/join/${params.get('joinToken')}`);
+      } catch (error) {
+        if (error && error.response && error.response.data && error.response.data.message) {
+          this.props.createErrorNotification(`Failed to accept invite: ${error.response.data.message}`);
+        } else {
+          this.props.createErrorNotification('Failed to accept invitation');
```
Comment (pketh) [inline src/presenters/pages/team.jsx]: for this and line `288` could we use something in natural language like 
> Invite failed, try asking your teammate to resend the invite
I'm assuming here that the main reason for a failed invite would be a old/expired join token

File: src/presenters/pages/team.jsx [hunk 2]
```diff
@@ -266,6 +269,47 @@ TeamPage.propTypes = {
...
+        }
+      }
+      this.setState({reload: true});
+    }
+    this.setState({pending: false});
+  }
+  
+  render() {
+    if (this.state.reload) {
+      return <Redirect to={this.props.location.pathname}/>;
```
Comment (pketh) [inline src/presenters/pages/team.jsx]: how does it feel irl? if it's not _too_ janky maybe it's fine for now. In theory, this was the nice thing about global models, where when the model for a team updates the right things change on a team

Comment (pketh) [reply src/presenters/pages/team.jsx]: (i'm having trouble with a model thing on the editor rn, so that's far from perfect too)

Comment (pketh) [reply src/presenters/pages/team.jsx]: should we show the user who just joined a new team , some kind of notification that they're not on the team? one way we could do that is a redirect on 200 with a diff queryString for joinSuccess, and then show a success notification when we see that

**Comment by pketh** [comment]
Actually there might be a third way, I can update the email template to send the user to a new community route that serves an extremely simple page with ‘joining teamname...’ then we can do the call there and then just load the team page. That might create an overall less janky experience.

File: src/presenters/pages/team.jsx (hunk 1 above)
Comment (GregWeil) [reply src/presenters/pages/team.jsx]: Sure, I'll keep the specific reason in the console

File: src/presenters/pages/team.jsx (hunk 2 above)
Comment (GregWeil) [reply src/presenters/pages/team.jsx]: Externally it's not too bad. The only visible thing is that the loader kind of flickers when it switches from accepting the invite to actually loading the team. (It doesn't actually blink out or anything, but the animation resets) The redirect is single page app, so we can create a success notification when we redirect and it'll stay around.

**Comment by pketh** [comment]
ok i'll update the email template later today to point to `/join/@:name/:code`

**Comment by GregWeil** [comment]
Updated to work as a separate /join/@team/token route

**Pull request merged**